inst/@sym/interval.m

```diff
@@ -0,0 +1,85 @@
+%% Copyright (C) 2016 Colin B. Macdonald and Lagu
+%%
+%% This file is part of OctSymPy.
+%%
+%% OctSymPy is free software; you can redistribute it and/or modify
+%% it under the terms of the GNU General Public License as published
+%% by the Free Software Foundation; either version 3 of the License,
+%% or (at your option) any later version.
+%%
+%% This software is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty
+%% of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @deftypefn  {Function File} {@var{r} =} interval (@var{A}, @var{B})
+%% @deftypefnx {Function File} {@var{r} =} interval (@var{A}, @var{B}, @var{lopen})
+%% @deftypefnx {Function File} {@var{r} =} interval (@var{A}, @var{B}, @var{lopen}, @var{ropen})
+%% Return an interval.
+%%
+%% Examples:
+%% @example
+%% @group
+%% interval(sym(0), sym(1))
+%%   @result{} (sym) [0, 1]
+%% interval(sym(0), 1, true, true)
+%%   @result{} (sym) (0, 1)
+%% interval(sym(0), 1, false, true)
+%%   @result{} (sym) [0, 1)
+%% @end group
+%% @end example
+%%
+%% Intervals can be degenerate:
+%% @example
+%% @group
+%% interval(sym(1), 1)
+%%   @result{} (sym) {1}
+%% interval(sym(2), 1)
+%%   @result{} (sym) ∅
+%% @end group
+%% @end example
+%%
+%% @seealso{union, intersect, setdiff, unique, ismember}
+%% @end deftypefn
+
+%% Author: Colin B. Macdonald
+%% Keywords: symbolic
+
+function I = interval(varargin)
+
+  if (nargin < 1 || nargin > 2)
+    print_usage();
+  end
+
+  varargin = sym(varargin);
+
+  % FIXME: Maybe in the future SymPy support more than 2 dimensions.
+
+  if nargin == 1
+
+    I = python_cmd ('return Interval(_ins[0], _ins[0]),', varargin{:});
+
+  else
```

Comment:
mm, i think can help to works because the out of some outputs are
```
c = (sym) {2}
```
so at least to me make sense can write ```interval(2)```, i'm thinking if we delete this part the user should start handling this from octave while its easly from here. and it feels the concepts don't match spliting this.